WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserAdapter.java

```diff
@@ -0,0 +1,198 @@
+        themeViewHolder.frameLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (isCurrent) {
+                    mCallback.onTryAndCustomizeSelected(themeId);
+                } else {
+                    mCallback.onViewSelected(themeId);
+                }
+            }
+        });
+    }
+
+    private void configureImageButton(Context context, ThemeViewHolder themeViewHolder, final String themeId, final boolean isPremium, boolean isCurrent) {
+        final PopupMenu popupMenu = new PopupMenu(context, themeViewHolder.imageButton);
+        popupMenu.getMenuInflater().inflate(R.menu.theme_more, popupMenu.getMenu());
+
+        configureMenuForTheme(popupMenu.getMenu(), isCurrent);
+
+        popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+            @Override
+            public boolean onMenuItemClick(MenuItem item) {
+                switch (item.getItemId()) {
+                    case R.id.menu_activate:
+                        if (isPremium) {
+                            mCallback.onDetailsSelected(themeId);
+                        } else {
+                            mCallback.onActivateSelected(themeId);
+                        }
+                        break;
+                    case R.id.menu_try_and_customize:
+                        mCallback.onTryAndCustomizeSelected(themeId);
+                        break;
+                    case R.id.menu_view:
+                        mCallback.onViewSelected(themeId);
+                        break;
+                    case R.id.menu_details:
+                        mCallback.onDetailsSelected(themeId);
+                        break;
+                    case R.id.menu_support:
+                    default:
+                        mCallback.onSupportSelected(themeId);
+                        break;
+                }
+
+                return true;
+            }
+        });
+        themeViewHolder.imageButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                popupMenu.show();
+            }
+        });
+    }
+
+    private void configureMenuForTheme(Menu menu, boolean isCurrent) {
+        MenuItem activate = menu.findItem(R.id.menu_activate);
+        MenuItem customize = menu.findItem(R.id.menu_try_and_customize);
+        MenuItem view = menu.findItem(R.id.menu_view);
+
+        if (activate != null) {
+            activate.setVisible(!isCurrent);
+        }
+        if (customize != null) {
+            if (isCurrent) {
+                customize.setTitle(R.string.customize);
+            } else {
+                customize.setTitle(R.string.theme_try_and_customize);
+            }
+        }
+        if (view != null) {
+            view.setVisible(!isCurrent);
+        }
+    }
+
+    static class ScreenshotHolder {
```

Comment:
Looks like this was going to be a bit more in-depth at some point, but I think we can just use `String` in the two places this is used.